cmd/quota.go

```diff
@@ -0,0 +1,108 @@
+package cmd
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	k8sResource "k8s.io/apimachinery/pkg/api/resource"
+)
+
+type quotaFlags struct {
+	cpuArg    string
+	cpu       k8sResource.Quantity
+	memoryArg string
+	memory    k8sResource.Quantity
+}
+
+func (quota *quotaFlags) parseQuotaFlags() {
+	if len(quota.cpuArg) != 0 {
+		quantity, err := k8sResource.ParseQuantity(quota.cpuArg)
+		if err != nil {
+			log.Error("Wrong cpu argument: " + err.Error())
+			return
+		}
+		quota.cpu = quantity
+	}
+
+	if len(quota.memoryArg) != 0 {
+		quantity, err := k8sResource.ParseQuantity(quota.memoryArg)
+		if err != nil {
+			log.Error("Wrong cpu argument: " + err.Error())
+			return
+		}
+		quota.memory = quantity
+	}
+}
+
+func checkQuota(container Container, quota quotaFlags, result *Result) {
+	if container.Resources.Limits == nil {
+		occ := Occurrence{id: ErrorResourcesLimitsNIL, kind: Warn, message: "Quota not set, please set it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+		return
+	}
+
+	checkCPULimit(container, quota, result)
+	checkMemoryLimit(container, quota, result)
+}
+
+func checkCPULimit(container Container, quota quotaFlags, result *Result) {
+	cpu := container.Resources.Limits.Cpu()
+	if cpu == nil || cpu.IsZero() {
+		occ := Occurrence{id: ErrorResourcesLimitsCpuNIL, kind: Warn, message: "CPU quota not set, please set it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+		return
+	}
+
+	if quota.cpu.MilliValue() > 0 && cpu.MilliValue() > quota.cpu.MilliValue() {
+		occ := Occurrence{id: ErrorResourcesLimitsCpuExceeded, kind: Warn, message: "CPU limit exceeded, please adjust it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+	}
+}
+
+func checkMemoryLimit(container Container, quota quotaFlags, result *Result) {
+	memory := container.Resources.Limits.Memory()
+	if memory == nil || memory.IsZero() {
+		occ := Occurrence{id: ErrorResourcesLimitsMemoryNIL, kind: Warn, message: "Memory quota not set, please set it!"}
```

Comment:
Why is this called `quota` here? Because what we seem to check from the code is `container.Resources.Limits.Memory()`